src/expressions/loop-expr.md

```diff
@@ -118,6 +138,37 @@ for n in 1..11 {
 assert_eq!(sum, 55);
 ```
 
+A for loop is equivalent to the following block expression.
+
+```rust
+'label: for PATTERN in iter_expr {
+    /* loop body */
+}
+```
+
+is equivalent to
+
+```rust
+{
+    let result = match IntoIterator::into_iter(iter_expr) {
```

Comment:
IIRC there should be some note here about how `into_iter` is looked up? It's *not* a lang item.